pages/searching/csv_export.rst

```diff
@@ -1,17 +1,55 @@
+This will open a modal where you can choose the fields the CSV document should contain. The initial fields are based on the message table configured for the search page.
+It is also possible to define a limit. Because messages are loaded in chunks, all chunks up to the one where the limit is reached will be retrieved.
+Which means the total number of delivered messages can be higher than the defined limit.
 
-.. image:: /images/searching/csv_export_dialog.png
+Clicking on *Start Download* will create the file based on the specified options and start the download.
+
+.. image:: /images/searching/csv_export_dialog_v2.png
+   :align: center
+
+When the search page or saved search contains multiple message tables, you can decide which message table you want to adopt the settings from.
+In this case you will see the following select when opening the CSV export modal:
+
+.. image:: /images/searching/csv_export_widget_select.png
    :align: center
 
-**Hint**: Some Graylog inputs keep the original message in the ``full_message`` field. If you need to export the original message, you
-can do so by clicking on the *List all fields* link at the bottom of the sidebar, and then selecting the ``full_message`` field.
+Additionally, you can select *Export to CSV* in a message table’s action menu, if you want to adopt its settings directly.
+
+.. image:: /images/searching/csv_export_message_table_action_menu.png
+   :align: center
+
+Exporting Message Tables on a Dashboard
+=======================================
+
+The export on a dashboard is slightly different compared to the search page or saved searches.
+Because dashboards do not have a single result set, you always need to select a message table when clicking on *Export to CSV* in a dashboard’s search bar action menu.
+You can find more information on this topic in the section :ref:`dashboards-widget-specific-criteria`.
+
+Even though a message table on dashboard has its own search criteria and the dashboard’s search bar only functions as a filter, you will always export the result set currently displayed by the message table.
+If you want to export only the search results the message table’s search criteria is referring to, make sure the dashboard’s filter is not defined.
+
+Decorator Support
+=================
+
+While the CSV export supports fields created by decorators, they are currently not being listed in the fields select options list.
+When you want to export a decorated field, enter its name in the field select and click on the option *Create field_name*.
+To ensure a decorated field is available in the context of the current search, open a message table's edit modal, by clicking on *Edit* in its context menu and find the decorators in the left sidebar.
+
+Exporting the full message
+==========================
+
+If you want to export the original message, keep in mind some Graylog inputs store the original message in the ``full_message`` field.
```

Comment:
If you want to export the full original message, keep in mind that it must be present in the stored message. Some Graylog inputs and file shippers can be configured to store the original message in the ``full_message`` field. Often, the message field may be used to export the entire, unparsed message.

Reply:
Thanks a lot @ChristopherKB!

Reply:
Yes, thanks @ChristopherKB! I've updated the page accordingly.